gateway/utils/utils.go

```diff
@@ -43,3 +50,15 @@ func ArrayContains(array []interface{}, value interface{}) bool {
 	}
 	return false
 }
+
+//ExtractRequestParams extract request info from http request & stores it in reqParam variable
+func ExtractRequestParams(r *http.Request, reqParams *model.RequestParams, body interface{}) {
```

Comment:
Let not use a pointer. Accept a struct and return the struct. Also, you haven't passed the payload at several places where you should have. Have a look at that again in all handlers

Reply:
Ok, I have checked all the config endpoint with POST method, there was not change.